php/libraries/NDB_Menu_Filter.class.inc

```diff
@@ -666,5 +666,42 @@ class NDB_Menu_Filter extends NDB_Menu
         $html = $smarty->fetch("menu_$this->menu.tpl");
         return $html;
     }
+
+    /**
+     * Returns the full list of candidates, users, etc.
+     *
+     * @return array
+     * @access private
+     */
+    function _getFullList()
+    {
+        // create DB object
+        $DB =& Database::singleton();
```

Comment:
You should use the factory so that this can be unit tested.

(It should also have a unit test..)